tests/rules/test_hadoop_dfs_missing_dash.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+from thefuck.rules.hadoop_dfs_missing_dash import match, get_new_command
+from tests.utils import Command
+
+
+@pytest.mark.parametrize('command', [
+    Command(script='./bin/hdfs dfs ls', stderr='ls: Unknown command\nDid you mean -ls?  This command begins with a dash.'),
```

Comment:
The command gives a really nice error message: it has the faulty command and the fix for it. Maybe you could make your rule more generic? You can get inspiration from the [*git_not_command*](https://github.com/nvbn/thefuck/blob/master/thefuck/rules/git_not_command.py) rule.